src/_pytest/cacheprovider.py

```diff
@@ -476,6 +476,14 @@ def pytest_sessionfinish(self) -> None:
 
 
 def pytest_addoption(parser: Parser) -> None:
+    """Add command-line options for cache functionality.
+
+    Adds options for last-failed (--lf), failed-first (--ff), new-first (--nf),
+    cache display (--cache-show), cache clearing (--cache-clear), and
+    last-failed-no-failures behavior (--lfnf). Also adds cache_dir ini option.
+
```

Comment:
I think this is describing the implementation in too much detail, so chances are this will get out of sync with the code quickly as we change/introduce options in the future. Better remove it.

```suggestion
```